pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,334 @@
+        """
+        log.info(f"Executing the shutdown function with ramp: {ramp} V/s.")
+
+        self.ramp_to_zero(ramp)
+        super().shutdown()
+
+    def ramp_to_zero(self, ramp):
+        """
+        Sets the voltage output setting to zero and the ramp setting
+        to a determinated value done by the ramp parameter.
+        In summary, the method conducts (ramps) the voltage output to zero
+        at a determinated voltage changing rate (ramp in V/s).
+        :param ramp: Is the changing rate (ramp in V/s) for the ramp setting
+        """
+        log.info("Executing the ramp_to_zero function with ramp: {ramp} V/s.")
+
+        self.voltage_ramp = ramp
+        time.sleep(self.COMMAND_DELAY)
+        self.voltage = 0
+        time.sleep(self.COMMAND_DELAY)
+        self.voltage = 0
+        time.sleep(self.COMMAND_DELAY)
+
+    def wait_for_voltage_output_set(self, check_period=1.0, timeout=60.0):
+        """
+        Waits until HV voltage output reaches the voltage ouput setting.
+        Checks the voltage output every check_period seconds and raises an exception
+        if the voltage output doesn't reach the voltage setting until the timeout time.
+        :param check_period: voltage output will be measured every check_period (seconds) time
+        :param timeout: time (seconds) give to the voltage output to reach the voltage setting
+        :return: None
+        :raises: Exception if the voltage output can't reach the voltage setting
+        before the timeout completes (seconds)
+        """
+        log.info("Executing the wait_for_voltage_output_set function.")
+
+        ref_time = time.time()
+        future_time = ref_time + timeout
+
+        voltage_output_setting = self.voltage
+        time.sleep(self.COMMAND_DELAY)
+        error = 0.02  # error of +-ten volts (0.01kV) to enter into the voltage stable window
+        voltage_output = self.measure_voltage
+        voltage_output_set = (voltage_output > (voltage_output_setting - error)) and \
+                             (voltage_output < (voltage_output_setting + error))
+
+        while not voltage_output_set:
+            actual_time = time.time()
+            log.debug(f"\tWaiting for voltage output set. "
+                      f"Reading output voltage every {check_period} seconds.\n"
+                      f"\tTimeout: {timeout} seconds. Elapsed time: "
+                      f"{round(actual_time - ref_time, ndigits=1)} seconds.")
+
+            time.sleep(check_period)  # wait for voltage output reaches the voltage output setting
+            voltage_output = self.measure_voltage
+            voltage_output_set = (voltage_output > (voltage_output_setting - error)) and \
+                                 (voltage_output < (voltage_output_setting + error))
+            log.debug("voltage_output_valid_range: "
+                      "[" + str(voltage_output_setting - error) +
+                      ", " + str(voltage_output_setting + error) + "]")
```

Comment:
as above, why emit this every iteration?

Reply:
In debug mode is important tracking the the voltage level at the out because sometimes the instrument miss the voltage_setting command. Also is important to show the error window. Thanks.